test/11_array.jl

```diff
@@ -531,9 +534,14 @@ for isordered in (false, true)
             cca2 = compact(ca2)
             r = vcat(cca1, cca2)
             @test r == vcat(a1, a2)
+            @test isa(cca1, CategoricalArray{Int,1,UInt8})
+            @test isa(cca2, CategoricalArray{Int,1,UInt8})
             @test isa(r, CategoricalArray{Int,1,CategoricalArrays.DefaultRefType})
             @test isa(vcat(cca1, ca2), CategoricalArray{Int,1,CategoricalArrays.DefaultRefType})
+            @test ordered(r) == false
+            @test sort(levels(r)) == collect(3:300)
```

Comment:
Well, it's less of an issue if the order of levels changes, but we should still return them in a meaningful and stable order. So better test that. In general it's a good idea to test everything, and possibly adapt the tests if we decide to change behaviour at some point.